packages/liferay-npm-scripts/src/scripts/lint.js

```diff
@@ -22,6 +23,159 @@ const DEFAULT_OPTIONS = {
+function findProjectIgnoreFiles(root) {
+	const cwd = `current working directory (${process.cwd()})`;
+
+	if (!root) {
+		log(`Unable to find "modules/" root from ${cwd}`);
+		return [];
+	}
+
+	const apps = path.join(root, 'apps');
+
+	try {
+		if (!fs.statSync(apps).isDirectory()) {
+			log(`"modules/apps/" in ${cwd} is not a directory`);
+			return [];
+		}
+	} catch (_error) {
+		log(`Unable to find "modules/apps/" from ${cwd}`);
+		return [];
+	}
+
+	const entries = fs.readdirSync(apps);
+
+	return entries.reduce((acc, entry) => {
+		const directory = path.join(apps, entry);
+
+		if (fs.statSync(directory).isDirectory()) {
+			const projects = fs.readdirSync(directory);
+
+			projects.forEach(project => {
+				const ignore = path.join(directory, project, '.eslintignore');
+
+				if (fs.existsSync(ignore)) {
+					acc.push(ignore);
+				}
+			});
+		}
+
+		return acc;
+	}, []);
+}
+
+/**
+ * ESLint only allows for a single .eslintignore file, and only in the current
+ * directory. In liferay-portal, however, we want the flexibility to have a
+ * top-level .eslintignore under "modules/apps/" and then project-specific ones
+ * that apply when running from a project subdirectory.
+ *
+ * When running from the modules root, we start with the top-level .eslintignore
+ * file and concatenate all sub-project .eslintignore files after relativizing
+ * the paths appropriately.
+ *
+ * When running from a specific project we just prepend the top-level
+ * .eslintignore, without peforming any relativization.
+ *
+ * @see https://eslint.org/docs/user-guide/configuring
+ */
+function getIgnoreFilePath() {
+	const ignores = fs.existsSync('.eslintignore')
+		? [fs.readFileSync('.eslintignore').toString()]
+		: [];
+
+	const root = findRoot();
+
+	if (process.cwd() === root) {
+		findProjectIgnoreFiles(root).forEach(file => {
+			const directory = path.dirname(file);
+			const lines = fs
+				.readFileSync(file)
+				.toString()
+				.split(/[\r\n]+/);
+
+			// Make patterns relative to the "modules/" root.
+			const patterns = lines.map(line => {
+				const match = line.match(/^\s*(\S.+)$/);
+
+				if (match) {
+					if (match[1].startsWith('#')) {
+						// Comment.
+						return line;
+					} else if (match[1].startsWith('!')) {
+						// Negated pattern.
+						return (
+							'!' + path.join('/', directory, match[1].slice(1))
+						);
+					} else {
+						// Normal pattern.
+						return path.join('/', directory, match[1]);
+					}
+				} else {
+					return line;
+				}
+			});
+
+			ignores.push(patterns.join('\n'));
+		});
+	} else if (root) {
+		// Prepend "modules/.eslintignore"
+		const rootIgnore = path.join(root, '.eslintignore');
+
+		if (fs.existsSync(rootIgnore)) {
+			ignores.unshift(fs.readFileSync(rootIgnore).toString());
+		} else {
+			log(`Did not find expected .eslintignore at ${root}`);
+		}
+	}
+
+	const contents = ignores.join('\n');
+
+	const directory = fs.mkdtempSync(
+		path.join(os.tmpdir(), 'liferay-npm-scripts-')
+	);
```

Comment:
Note: explicitly not cleaning up this file because the OS will reclaim anything we leave in `os.tmpdir()` at a later time, which might be convenient for inspection, and we don't have to worry about it cluttering up the `git status`.

Reply:
I never knew about `os.tmpdir()` before. Very cool.